internal/graph/singleflight_resolver.go

```diff
@@ -53,6 +54,11 @@ func (s *singleflightCheckResolver) ResolveCheck(
 		return nil, err
 	}
 
+	ctx, span := tracer.Start(ctx, "ResolveCheck")
+	defer span.End()
+	span.SetAttributes(attribute.String("resolver_type", "SingleflightCheckResolver"))
+	span.SetAttributes(attribute.String("tuple_key", req.GetTupleKey().String()))
+
```

Comment:
Why is this new span being created after the `if`? in the Cached resolver, it's right at the start. 


If you put it at the start, it gives you the opportunity of marking the spans as errored if there was an error.

```go
        ctx, span := tracer.Start(ctx, "ResolveCheck")
	defer span.End()
	span.SetAttributes(attribute.String("resolver_type", "SingleflightCheckResolver"))
	span.SetAttributes(attribute.String("tuple_key", req.GetTupleKey().String()))
	
	key, err := CheckRequestCacheKey(req)
	if err != nil {
		telemetry.TraceError(span, err) // <------ this line
		s.logger.Error("singleflight cache key computation failed with error", zap.Error(err))
		return nil, err
	}
```

That aside, we should also be doing `telemetry.TraceError` here:

```go
    singleFlightResp, err, shared := s.group.Do(key, func() (interface{}, error) {
		// ...
	})
	if err != nil {
		telemetry.TraceError(span, err)
		return nil, err
	}
```

So that this

<img width="1329" alt="image" src="https://github.com/openfga/openfga/assets/5374887/d4420b75-0213-44e5-9cdb-ae107eadb1ce">

becomes this

<img width="1366" alt="image" src="https://github.com/openfga/openfga/assets/5374887/5436381c-7fbe-4547-bcfb-1ac578e9eaf0">

Reply:
No particular reason for placement but great suggestion, will change.